tools/compile_tf_graph.py

```diff
@@ -568,21 +695,48 @@ def _get_parent_layer(self, layer_name):
       return layer
     assert isinstance(rec_layer, RecStepByStepLayer)
     output = layer.output.copy()
-    output.placeholder = rec_layer.create_state_var(
-      name="base_value_%s" % layer_name, initial_value=output.placeholder, data_shape=output)
+    output = self.make_base_state_vars(layer_name, output)
+    layer = WrappedInternalLayer(name=layer_name, network=self.parent_net, output=output, base_layer=layer)
+    self._parent_layers[layer_name] = layer
+    return layer
+
+  # make state vars for input from outside the loop (automatic tile_batch)
+  def make_base_state_vars(self, name, output):
+    rec_layer = self.parent_rec_layer
+    from TFUtil import tile_transposed
+    output.placeholder = tile_transposed(
+      rec_layer.create_state_var(name="base_value_%s" % name, initial_value=output.placeholder, data_shape=output),
+      axis=output.batch_dim_axis, 
+      multiples=self.parent_tile_multiples_var.read_value())
     from TFUtil import DimensionTag
     for i, size in list(output.size_placeholder.items()):
       dim_tag = DimensionTag.get_tag_from_size_tensor(size)
       if not dim_tag:
-        print("Warning, no defined dim tag on %r, axis %i" % (layer, output.get_batch_axis(i)), file=log.v2)
+        print("Warning, no defined dim tag on %r, axis %i" % (name, output.get_batch_axis(i)), file=log.v2)
         dim_tag = output.get_dim_tag(output.get_batch_axis(i))
         dim_tag.set_tag_on_size_tensor(size)
-      new_size = rec_layer.create_state_var(name="base_size%i_%s" % (i, layer_name), initial_value=size)
+      new_size = rec_layer.create_state_var(name="base_size%i_%s" % (i, name), initial_value=size)
+      new_size = tile_transposed(new_size, axis=0, multiples=self.parent_tile_multiples_var.read_value())
       dim_tag.set_tag_on_size_tensor(new_size)
       output.size_placeholder[i] = new_size
-    layer = WrappedInternalLayer(name=layer_name, network=self.parent_net, output=output, base_layer=layer)
-    self._parent_layers[layer_name] = layer
-    return layer
+    return output
+ 
+
```

Comment:
Why two empty lines? It should be one empty line here. (PyCharm should tell you that, or not?)